exercises/robot-name/example.rkt

```diff
@@ -0,0 +1,42 @@
+#lang racket
+(require racket/class)
+
+(provide robot%)
+
+(define (random-digit-char)
+  (integer->char
+   (random (char->integer #\0)
+           (add1 (char->integer #\9)))))
+
+(define (random-letter-char)
+  (integer->char
+   (random (char->integer #\A)
+           (add1 (char->integer #\Z)))))
+
+(define (generate-valid-name)
+  (let ([all-robot-names (set)])
+    (list->string
+     (let loop ([index 0])
+       (cond [(< index 2) (cons (random-letter-char) (loop (add1 index)))]
+             [(< index 6) (cons (random-digit-char) (loop (add1 index)))]
+             [else empty])))))
+
+(define generate-unique-name
+  (let ([all-robot-names (mutable-set)])
+    (lambda ()
+      (let ([name (generate-valid-name)])
+        (cond [(set-member? all-robot-names name)
+               (generate-unique-name)]
```

Comment:
I love that you are using a set here.

Reply:
My boss says that I look for excuses to use things like that.